Signal/src/util/OWSBackupAPI.swift

```diff
@@ -92,47 +106,46 @@ import CloudKit
     public class func saveRecordToCloud(record: CKRecord,
                                         success: @escaping (String) -> Void,
                                         failure: @escaping (Error) -> Void) {
-
-        let myContainer = CKContainer.default()
-        let privateDatabase = myContainer.privateCloudDatabase
-        privateDatabase.save(record) {
-            (record, error) in
-
-            if let error = error {
-                Logger.error("\(self.logTag) error saving record: \(error)")
-                failure(error)
-            } else {
-                guard let recordName = record?.recordID.recordName else {
-                    Logger.error("\(self.logTag) error retrieving saved record's name.")
-                    failure(OWSErrorWithCodeDescription(.exportBackupError,
-                                                        NSLocalizedString("BACKUP_EXPORT_ERROR_SAVE_FILE_TO_CLOUD_FAILED",
-                                                                          comment: "Error indicating the a backup export failed to save a file to the cloud.")))
-                    return
-                }
-                Logger.info("\(self.logTag) saved record.")
-                success(recordName)
-            }
-        }
+        saveRecordToCloud(record: record,
+                          remainingRetries: maxRetries,
+                          success: success,
+                          failure: failure)
     }
 
-    @objc
-    public class func deleteRecordFromCloud(recordName: String,
-                                            success: @escaping (()) -> Void,
-                                            failure: @escaping (Error) -> Void) {
+    private class func saveRecordToCloud(record: CKRecord,
+                                         remainingRetries: Int,
+                                         success: @escaping (String) -> Void,
+                                         failure: @escaping (Error) -> Void) {
 
-        let recordID = CKRecordID(recordName: recordName)
+        database().save(record) {
+            (_, error) in
 
-        let myContainer = CKContainer.default()
-        let privateDatabase = myContainer.privateCloudDatabase
-        privateDatabase.delete(withRecordID: recordID) {
-            (record, error) in
-
-            if let error = error {
-                Logger.error("\(self.logTag) error deleting record: \(error)")
-                failure(error)
-            } else {
-                Logger.info("\(self.logTag) deleted record.")
-                success()
+            let outcome = outcomeForCloudKitError(error: error,
```

Comment:
The new `outcomeForCloudKitError` method parses CloudKit responses and buckets them into "outcomes" that simplify how we handle them.  Handling them is still pretty repetitive, but I think DRY'ing this up would yield awkward, unreadable code.